packages/angular-query-experimental/src/QueryClientService.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable, inject } from '@angular/core'
+import { QUERY_CLIENT } from './queryClient'
+import type { OnDestroy } from '@angular/core'
+import type { QueryClient } from '@tanstack/query-core'
+
+/**
+ * Takes care of mounting and unmounting the query client.
+ * @internal
+ */
+@Injectable()
+export class QueryClientService implements OnDestroy {
+  readonly #queryClient: QueryClient | undefined
```

Comment:
```suggestion
  readonly #queryClient = inject(QUERY_CLIENT, { optional: true });
```

Reply:
We can just optionally inject it here, and in the constructor can mount it if it exists.